.github/workflows/update-doc.yaml

```diff
@@ -10,16 +10,14 @@ on:
     - hack/doc-site/**
     - .github/workflows/update-doc.yaml
 
-  pull_request_target:
+  pull_request:
```

Comment:
This change is done to make it easier to change and test the workflow in the PR's (for example now, when I need to show that this change actually works without merging it to master)

The `pull_request_target` workflows always run with the yaml from the master branch, the `pull_request` ones use the ones from the PR